### PR TITLE
Speedup testing [6501]

### PR DESCRIPTION
**Pull request by LuisGP**
* Added parallel build
* Check if Fast-RTPS was already built

### CODE REVIEW

File: src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java [hunk 1]
```diff
@@ -40,12 +42,19 @@ public void runTests()
 
         //Configure Fast-RTPS for the tests
         ArrayList<String[]> commands = new ArrayList<String[]>();
-        commands.add(new String[]{"mkdir -p " + OUTPUT_PATH, "."});
-        commands.add(new String[]{"rm -rf Fast-RTPS", OUTPUT_PATH});
-        commands.add(new String[]{"git clone -b " + branch + " https://github.com/eProsima/Fast-RTPS.git", OUTPUT_PATH});
-        commands.add(new String[]{"mkdir build", OUTPUT_PATH + "/Fast-RTPS"});
+        if (Files.notExists(Paths.get(OUTPUT_PATH + "/Fast-RTPS/build")))
```
Comment (raquelalvarezbanos) [inline src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java]: What if the path exists but using a branch different from the one we currently want? Would we be using the old one?

**Comment by TSC21** [comment]
Instead of cloning, why aren't we checking first if Fast-RTPS is installed on the system? Also, if Fast-RTPS is required for testing, IMHO it should be made as a dependency, not cloned on build time.

File: src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java (hunk 1 above)
Comment (LuisGP) [reply src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java]: Take a look at the `else` branch.

**Comment by LuisGP** [comment]
> Instead of cloning, why aren't we checking first if Fast-RTPS is installed on the system? Also, if Fast-RTPS is required for testing, IMHO it should be made as a dependency, not cloned on build time.

Typically, we never install Fast-RTPS system-wide. This way, we can test Fast-RTPS-Gen against a concrete Fast-RTPS branch, which would not be possible if we use an already installed Fast-RTPS.

As you said, Fast-RTPS is only required for testing, so keep Fast-RTPS-Gen without such dependency is cleaner, I think.

File: src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java (hunk 1 above)
Comment (raquelalvarezbanos) [reply src/test/java/com/eprosima/fastrtps/FastRTPSGenTest.java]: Sorry I completely missed that :(

**Pull request merged**